src/network/networkd-route.c

```diff
@@ -516,6 +518,65 @@ int route_expire_handler(sd_event_source *s, uint64_t usec, void *userdata) {
         return 1;
 }
 
+static int append_nexthop_one(Route *route, MultipathRoute *m, struct rtattr *rta, size_t max_length, struct rtnexthop *rtnh) {
+        int r;
+
+        *rtnh = (struct rtnexthop) {
+                .rtnh_len = sizeof(*rtnh),
+                .rtnh_ifindex = m->ifindex,
+                .rtnh_hops = m->weight > 0 ? m->weight - 1 : 0,
+        };
+
+        rta->rta_len += rtnh->rtnh_len;
+
+        if (route->family == m->gateway.family) {
+                r = rtattr_append_attribute(rta, max_length, RTA_GATEWAY, &m->gateway.address, FAMILY_ADDRESS_SIZE(m->gateway.family));
+                if (r < 0)
+                        return r;
+                rtnh->rtnh_len += RTA_SPACE(FAMILY_ADDRESS_SIZE(m->gateway.family));
+        } else {
+                r = rtattr_append_attribute(rta, max_length, RTA_VIA, &m->gateway, FAMILY_ADDRESS_SIZE(m->gateway.family) + sizeof(m->gateway.family));
+                if (r < 0)
+                        return r;
+                rtnh->rtnh_len += RTA_SPACE(FAMILY_ADDRESS_SIZE(m->gateway.family) + sizeof(m->gateway.family));
+        }
+
+        return 0;
+}
+
+static int append_nexthops(Route *route, sd_netlink_message *req) {
+        struct rtnexthop *rtnh;
+        struct rtattr *rta;
+        MultipathRoute *m;
+        char buf[4096];
```

Comment:
where does this size come from? shouldn't this grow automatically (via GREEDY_REALLOC for example)